dast/lcs.py

```diff
@@ -4,43 +4,43 @@
 from collections import defaultdict
 from typing import Any, Dict, Iterable, List, Tuple
 
-def get_lcs(x: Iterable[Any], y: Iterable[Any]):
+def get_lcs(xs: Iterable[Any], ys: Iterable[Any]):
     """
     Longest common subsequence between two finite iterables via dynamic programming.
 
     >>> get_lcs("abcde", "abfgdfe")
     (['a', 'b', 'd', 'e'], [True, True, False, True, True], [True, True, False, False, True, False, True])
     """
-    all_x = list(x)
-    all_y = list(y)
-    num_x = len(all_x)
-    num_y = len(all_y)
-
     # sub_problems[i, j] = LCS between all_x[:i] and all_y[:j]
     sub_problems: Dict[Tuple[int, int], List] = defaultdict(list)
 
-    for i in range(num_x):
-        for j in range(num_y):
-            if all_x[i] == all_y[j]:
-                sub_problems[i + 1, j + 1] = sub_problems[i, j] + [all_x[i]]
+    for i, x in enumerate(xs):
+        for j, y in enumerate(ys):
+            if x == y:
+                sub_problems[i + 1, j + 1] = sub_problems[i, j] + [x]
             else:
                 sub_problems[i + 1, j + 1] = max(
                     sub_problems[i, j + 1],
                     sub_problems[i + 1, j],
                     key=len)
 
-    lcs = sub_problems[num_x, num_y]
+    lcs = sub_problems[len(xs), len(ys)]
```

Comment:
In general `Iterable`s do not support `len`, so maybe we should just change the signature to `List` instead?

Reply:
Indeed I fixed it in the follow up commit, we can even use i+1, j+1, but it would not be that explicit

Reply:
The diff looks funny though, still has come commits that are already in main